packages/trimerge-sync/src/TrimergeClient.ts

```diff
@@ -318,13 +338,29 @@ export class TrimergeClient<
     this.clientList = Array.from(this.clientMap.values());
     this.clientListSubs.emitChange(event);
   }
-  private sync(): void {
+  private async sync() {
     const commits = this.unsyncedCommits;
     if (commits.length > 0 || this.newPresence !== undefined) {
       this.unsyncedCommits = [];
       this.updateSyncState({ localSave: 'saving' });
-      this.store.update(commits, this.newPresence);
+      try {
+        await this.store.update(
+          commits.map((c) => c.commit),
+          this.newPresence,
+        );
+      } catch (e) {
+        for (const commit of commits) {
+          if (commit.reject) {
+            commit.reject(e);
```

Comment:
So, we're rejecting the promises for all of the commits if any of the `commits` failed?

Reply:
should be addressed by the restructure